sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collationExpressions.scala

```diff
@@ -90,6 +100,9 @@ case class Collate(child: Expression, collationName: String)
   since = "4.0.0",
   group = "string_funcs")
 case class Collation(child: Expression) extends UnaryExpression with RuntimeReplaceable {
+  if (!SQLConf.get.collationEnabled) {
```

Comment:
Could you add the parameter `collationEnabled` to the case class, and avoid the check in the constructor. This will avoid an issue fixed by this PR: https://github.com/apache/spark/pull/27658 (see `Why are the changes needed?`)

Reply:
It might be too much for a temporary config. Maybe let's add a new check rule in `BaseSessionStateBuilder.analyzer.extendedCheckRules`, and fail if these two expressions are found in the plan tree.